core/services/log/pool.go

```diff
@@ -0,0 +1,45 @@
+package log
+
+import (
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+type (
+	logPool struct {
+		allLogs []types.Log
+	}
+)
+
+func newLogPool() *logPool {
+	return &logPool{
+		allLogs: make([]types.Log, 0),
+	}
+}
+
+func (pool *logPool) addLog(log types.Log) {
+	pool.allLogs = append(pool.allLogs, log)
+}
+
+func (pool *logPool) getLogsToSend(head *models.Head, highestNumConfirmations uint64, finalityDepth uint64) []types.Log {
+	latestBlockNum := uint64(head.Number)
+	logsToReturn := pool.allLogs
+	logsToKeep := make([]types.Log, 0)
```

Comment:
nitpick, but the "0" in unnecessary, right?

Reply:
ah yes indeed